provider/cloudflare/cloudflare.go

```diff
@@ -826,16 +840,32 @@ func (p *CloudFlareProvider) newCloudFlareChange(action string, ep *endpoint.End
 			RegionKey: regionKey,
 		}
 	}
+
+	priority := (*uint16)(nil)
+	if ep.RecordType == "MX" {
+		parsedPriority, mailHost, err := parseMXRecord(target)
+		if err != nil {
+			log.Errorf("Failed to parse MX record target %q: %v", target, err)
+		} else {
+			priority = &parsedPriority
+			target = mailHost
+		}
+	}
+
 	return &cloudFlareChange{
 		Action: action,
 		ResourceRecord: cloudflare.DNSRecord{
 			Name: ep.DNSName,
 			TTL:  ttl,
 			// We have to use pointers to bools now, as the upstream cloudflare-go library requires them
 			// see: https://github.com/cloudflare/cloudflare-go/pull/595
-			Proxied: &proxied,
-			Type:    ep.RecordType,
-			Content: target,
+			Proxied:  &proxied,
+			Type:     ep.RecordType,
+			Content:  target,
+			Priority: priority,
+			Meta: map[string]interface{}{
+				"region": p.RegionKey,
```

Comment:
cc @vflaux for the regional hostnames related change.
(was it necessary to include it into this PR?)

Reply:
I don’t think it’s necessary since the Cloudflare api don't expect a `meta` field for Create or Update operations. The field is there for api calls that returns DNS records.

- Create : https://developers.cloudflare.com/api/resources/dns/subresources/records/methods/create/
- Record: https://developers.cloudflare.com/api/resources/dns/subresources/records/models/mx_record/#(schema)